doc/api/api_changes_3.3/behaviour.rst

```diff
@@ -186,3 +186,19 @@ anymore.
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 This ensures that they go above gridlines by default.  The old ``zorder`` can
 be kept by passing it as a keyword argument to `.Axes.hist`.
+
+tight/constrained_layout no longer worry about titles that are too wide
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+*tight_layout* and *constrained_layout* shrink axes to accommodate
+"decorations" on the axes.  However, if an xlabel or title is too long in the
+x direction, making the axes smaller in the x-direction doesn't help.  The
+behavior of both has been changed to ignore the width of the title and
+xlabel and the height of the ylabel in the layout logic.
+
+This also means there is a new keyword argument for `.axes.Axes.get_tightbbox`:
+``for_layout_only``, which defaults to *False*, but if *True* returns a
+bounding box using the rules above.  `.axis.Axis.get_tightbbox` gets an
+``ignore_label`` keyword argument, which is *None* by default, but which can
+also be 'x' or 'y'. These API changes are public, but are meant to be
+mostly used internally.
```

Comment:
You could make the kwargs `_for_layout_only`/`_ignore_label` with a leading underscore, depending on how private you want to make them...

Reply:
Yeah, thought about it, but its possible someone else would like the functionality?

Reply:
I'm fine if you want to declare this public API, I'm just not a fan of "mostly private, but not really."